samples/bluetooth/peripheral_ams_client/src/main.c

```diff
@@ -0,0 +1,505 @@
+	HAS_PREVIOUS_TRACK
+};
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_LIMITED | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_SOLICIT128, BT_UUID_AMS_VAL),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static struct bt_ams_client ams_c;
+
+static const enum bt_ams_player_attribute_id entity_update_player[] = {
+	BT_AMS_PLAYER_ATTRIBUTE_ID_NAME,
+	BT_AMS_PLAYER_ATTRIBUTE_ID_PLAYBACK_INFO,
+	BT_AMS_PLAYER_ATTRIBUTE_ID_VOLUME
+};
+
+static const enum bt_ams_track_attribute_id entity_update_track[] = {
+	BT_AMS_TRACK_ATTRIBUTE_ID_ARTIST,
+	BT_AMS_TRACK_ATTRIBUTE_ID_ALBUM,
+	BT_AMS_TRACK_ATTRIBUTE_ID_TITLE,
+	BT_AMS_TRACK_ATTRIBUTE_ID_DURATION
+};
+
+static atomic_t appl_flags;
+
+static char msg_buff[MAX_MSG_SIZE + 1];
+
+static void ea_read_cb(struct bt_ams_client *ams_c, uint8_t err, const uint8_t *data, size_t len)
+{
+	if (err) {
+		printk("AMS EA read error 0x%02X\n", err);
+	} else if (len > MAX_MSG_SIZE) {
+		printk("AMS EA data size is too big\n");
+	} else {
+		memcpy(msg_buff, data, len);
+		msg_buff[len] = '\0';
+		printk("AMS EA: %s\n", msg_buff);
+	}
+}
+
+static void ea_write_cb(struct bt_ams_client *ams_c, uint8_t err)
+{
+	if (!err) {
+		err = bt_ams_read_entity_attribute(ams_c, ea_read_cb);
+	} else {
+		printk("AMS EA write error 0x%02X\n", err);
+	}
+}
+
+static void notify_rc_cb(struct bt_ams_client *ams_c,
+			 const uint8_t *data, size_t len)
+{
+	size_t i;
+	char str_hex[4];
+	bool has_next_track;
+	bool has_previous_track;
+	enum bt_ams_remote_command_id cmd_id;
+
+	if (len > 0) {
+		if (len * 3 > MAX_MSG_SIZE) {
+			printk("AMS RC data size is too big\n");
+		} else {
+			/* Print the accepted Remote Command values. */
+			sprintf(msg_buff, "%02X", *data);
+
+			for (i = 1; i < len; i++) {
+				sprintf(str_hex, ",%02X", *(data + i));
+				strcat(msg_buff, str_hex);
+			}
+
+			printk("AMS RC: %s\n", msg_buff);
+		}
+
+	}
+
+	/* Check if track commands are available. */
+	has_next_track = has_previous_track = false;
+	for (i = 0; i < len; i++) {
+		cmd_id = (enum bt_ams_remote_command_id)*(data + i);
+		if (cmd_id == BT_AMS_REMOTE_COMMAND_ID_NEXT_TRACK) {
+			has_next_track = true;
+		} else if (cmd_id == BT_AMS_REMOTE_COMMAND_ID_PREVIOUS_TRACK) {
+			has_previous_track = true;
+		}
+	}
+	atomic_set_bit_to(&appl_flags, HAS_NEXT_TRACK, has_next_track);
+	atomic_set_bit_to(&appl_flags, HAS_PREVIOUS_TRACK, has_previous_track);
+}
+
+static void notify_eu_cb(struct bt_ams_client *ams_c,
+			 const struct bt_ams_entity_update_notif *notif,
+			 int err)
+{
+	uint8_t attr_val;
+	char str_hex[9];
+
+	if (!err) {
+		switch (notif->ent_attr.entity) {
+		case BT_AMS_ENTITY_ID_PLAYER:
+			attr_val = notif->ent_attr.attribute.player;
+			break;
+		case BT_AMS_ENTITY_ID_QUEUE:
+			attr_val = notif->ent_attr.attribute.queue;
+			break;
+		case BT_AMS_ENTITY_ID_TRACK:
+			attr_val = notif->ent_attr.attribute.track;
+			break;
+		default:
+			err = -EINVAL;
+		}
+	}
+
+	if (err) {
```

Comment:
You could check error at the beginning of this function which could simplify the logic of this function a little.

Reply:
Could you elaborate? I cannot figure out how to procceed.